### PR TITLE
feat(namespace): add APPLICATIONS_NAMESPACE env var support to GetNamespace()

### PR DESCRIPTION
**Pull request by kpunwatk**
Related to [RHOAIENG-67659](https://redhat.atlassian.net/browse/RHOAIENG-67659)
Context:
In modular deployments, the ODH platform operator injects APPLICATIONS_NAMESPACE to indicate where operands should be deployed. This change enables the trustyai-service-operator to read this env var for namespace discovery.

Changes:
- Update GetNamespace() to check APPLICATIONS_NAMESPACE env var first
- Fall back to service account namespace file when env var is unset
- Add unit tests covering both resolution paths (4/4 specs passing)
- Document the env-first resolution order in function comment

Resolution order:
1. APPLICATIONS_NAMESPACE environment variable (platform-injected)
2. /var/run/secrets/kubernetes.io/serviceaccount/namespace (fallback)

This implements [GAP-03](https://redhat.atlassian.net/browse/GAP-03) from [RHOAIENG-66846](https://redhat.atlassian.net/browse/RHOAIENG-66846) (modular architecture migration).

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved namespace detection to prioritize the `APPLICATIONS_NAMESPACE` environment variable when set.
  * When `APPLICATIONS_NAMESPACE` is not set, the app falls back to the Kubernetes service account namespace file.
  * This makes namespace selection more predictable across environments.

* **Tests**
  * Added/updated automated coverage for namespace resolution precedence and fallback behavior.

### CODE REVIEW

File: controllers/utils/utils_test.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestUtils(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Utils Suite")
+}
```
[the remark this answers is not in the document]
Comment (kpunwatk) [reply controllers/utils/utils_test.go:15]: This is a pure utility function test that validates OS-level behavior (environment variables and file I/O), not Kubernetes API interactions.
  
  What GetNamespace() does:
  1. Reads APPLICATIONS_NAMESPACE environment variable
  2. Falls back to reading /var/run/secrets/kubernetes.io/serviceaccount/namespace file
  3. Returns a string
  
  No Kubernetes API calls → No need for envtest

File: controllers/utils/utils_test.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+			// The actual service account file doesn't matter since env var takes precedence
+			ns, err := GetNamespace()
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(ns).To(Equal(envNS))
+		})
+	})
+
+	Context("when APPLICATIONS_NAMESPACE environment variable is empty", func() {
```
Comment (ruivieira) [inline controllers/utils/utils_test.go:75]: @kpunwatk would it be ok to add a test for when `APPLICATIONS_NAMESPACE` is empty _and_ the service account file doesn't exist?

**Review by ruivieira — APPROVED** [review]

File: controllers/utils/utils_test.go (hunk 2 above)
Comment (kpunwatk) [reply controllers/utils/utils_test.go:75]: addressed [edc482e](https://github.com/trustyai-explainability/trustyai-service-operator/pull/784/commits/edc482e047c4613c6e3d6e95bf76c955497d281c)
